envoy/stream_info/stream_info.h

```diff
@@ -758,6 +773,16 @@ class StreamInfo {
    * This function is called on Envoy fatal errors so should avoid memory allocation.
    */
   virtual void dumpState(std::ostream& os, int indent_level = 0) const PURE;
+
+  /**
+   * @param deferredLoggingInfo request/response info for deferred logging.
```

Comment:
nits: s/info/headers and trailers

How about changing this interface to be `storeDeferredLoggignInfo(request_header_map_shard_ptr, response_header_map_shard_ptr, trailer_map_shard_ptr)` so that ActiveStream::deferredLoggingInfo() is no longer needed.

Reply:
I added ActiveStream::deferredLoggingInfo() to avoid having to create new getters for each of the shared pointers. There are getters for the raw pointers here: https://github.com/envoyproxy/envoy/blob/main/source/common/http/conn_manager_impl.h#L237-L251, but I'd need to add versions for each of the shared ones. LMK if you have any better ideas.

Reply:
I will change all of the "info"s to "headers and trailers".